feat(inference): migration v1 and support BYOM (#3048)
scaleway/terraform-provider-scaleway: docs/resources/inference_custom_model.md

---
subcategory: "Inference"

> page_title: "Scaleway: scaleway_inference_deployment"

Reviewer: ```suggestion
page_title: "Scaleway: scaleway_inference_custom_model"
```

---

# Resource: scaleway_inference_custom_model

> The scaleway_inference_custom_model resource allows you to upload and manage custom inference models in the Scaleway Inference ecosystem. Once registered, a custom model can be used in an scaleway_inference_deployment resource.

Reviewer 2: ```suggestion
The scaleway_inference_custom_model resource allows you to upload and manage custom inference models in the Scaleway Inference ecosystem. Once registered, a custom model can be used in any scaleway_inference_deployment resource.
```

- `project_id` - (Defaults to [provider](../index.md#project_id) `project_id`) The ID of the project the deployment is associated with.

## Attributes Reference

In addition to all arguments above, the following attributes are exported:

- `id` - The unique identifier of the custom model.
- `tags` - Tags associated with the model.
- `status` - The current status of the model (e.g., ready, error, etc.).
- `description` - A textual description of the model (if available).
- `has_eula` - Whether the model requires end-user license agreement acceptance before use.
- `parameter_size_bits` - Size, in bits, of the model parameters.
- `size_bytes` - Total size, in bytes, of the model archive.
- `nodes_support` - List of supported node types and their quantization options. Each entry contains:
        - `node_type_name` - The type of node supported.
        - `quantization` - A list of supported quantization options, including:
            - `quantization_bits` -  Number of bits used for quantization (e.g., 8, 16).
            - `allowed` - Whether this quantization is allowed.

>             - `max_context_size` - Maximum context length supported by this quantization.

Reviewer: Import part is missing

---
subcategory: "Inference"

> page_title: "Scaleway: scaleway_inference_custom_model"

Reviewer 3: Why not model?

---

# Resource: scaleway_inference_custom_model

> The scaleway_inference_custom_model resource allows you to upload and manage custom inference models in the Scaleway Inference ecosystem. Once registered, a custom model can be used in any scaleway_inference_deployment resource.

Reviewer 3: Why not `scaleway_inference_model`?

- `project_id` - (Defaults to [provider](../index.md#project_id) `project_id`) The ID of the project the deployment is associated with.

## Attributes Reference

In addition to all arguments above, the following attributes are exported:

- `id` - The unique identifier of the custom model.
- `tags` - Tags associated with the model.
- `status` - The current status of the model (e.g., ready, error, etc.).
- `description` - A textual description of the model (if available).
- `has_eula` - Whether the model requires end-user license agreement acceptance before use.
- `parameter_size_bits` - Size, in bits, of the model parameters.
- `size_bytes` - Total size, in bytes, of the model archive.
- `nodes_support` - List of supported node types and their quantization options. Each entry contains:
        - `node_type_name` - The type of node supported.
        - `quantization` - A list of supported quantization options, including:
            - `quantization_bits` -  Number of bits used for quantization (e.g., 8, 16).
            - `allowed` - Whether this quantization is allowed.
            - `max_context_size` - Maximum context length supported by this quantization.

## Import

> Functions can be imported using, `{region}/{id}`, as shown below:

Reviewer: ```suggestion
Custom models can be imported using, `{region}/{id}`, as shown below:
```
